controllers/modelmesh/modelmesh.go

```diff
@@ -47,6 +47,7 @@ type Deployment struct {
 	Metrics            bool
 	PrometheusPort     uint16
 	PrometheusScheme   string
+	PayloadProcessors  string
```

Comment:
I think it would be better for this to be a `string[]`, and then validate that none of the values contains a space.

Reply:
@njhill would the plan be then:

1) Make the modelmesh.PayloadProcessors a `string[]`
2) Have the `config.yaml `schema expect a list in the `PayloadProcessors` yaml field
3) When the `modelmesh.PayloadProcessors` struct is used to construct the `pkg/config` struct (i.e., [here](https://github.com/kserve/modelmesh-serving/pull/339/files#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5R339), unpack the array, check if there's no spaces in any item, and pass it to the config struct as a space-separated string?

Reply:
@RobGeada yes, exactly.

Kind of separate though but I do have a slight concern that the current URI spec for the processors might need more thought to ensure it's sufficiently generic, for example I imagine it may be common to want to enable for _all_ models not just specific ones, etc. But that doesn't really affect this PR.

Reply:
Sorry, hang on, shouldn't it be the other way around?

The `pkg/config` struct creates the `modelmesh.Deployment` struct, which is then used to populate the deployment template. Surely we want the `pkg/config` to accept the string[] from the yaml, and then pass it to the space separated string in the `modelmesh.Deployment` struct.

So:

- [pkg/config](https://github.com/RobGeada/modelmesh-serving/blob/ab45d5cbca172e28fcba31e18507da80071543c3/pkg/config/config.go#L79).PayloadProcessors -> string[], from yaml
- [controllers/modelmesh.Deployment.PayloadProcessors](https://github.com/RobGeada/modelmesh-serving/blob/PayloadProcessorEndpointConfig/controllers/modelmesh/modelmesh.go) -> string
- In [servingruntime_controller](https://github.com/RobGeada/modelmesh-serving/blob/ab45d5cbca172e28fcba31e18507da80071543c3/controllers/servingruntime_controller.go#L251), parse `pkg/config` string array into a space separated list